src/syrupy/report.py

```diff
@@ -63,9 +63,15 @@ class SnapshotReport:
     _invocation_args: Tuple[str, ...] = attr.ib(factory=tuple)
     _provided_test_paths: Dict[str, List[str]] = attr.ib(factory=dict)
     _keyword_expressions: Set["Expression"] = attr.ib(factory=set)
+    _collected_items_by_nodeid: Dict[str, "pytest.Item"] = attr.ib(
+        factory=dict, init=False
+    )
```

Comment:
This is generated once at instantiation, and used to easily map selected item nodes to collected pytest items.